required/tools/longtable.dtx

```diff
@@ -2206,12 +2206,16 @@
   \bgroup
 %    \end{macrocode}
 % \changes{v4.12}{2019/02/06}{Color group to match array package}
+% \changes{v4.18}{2023/08/25}{Use \cs{everypar} to match array package, github 1128}% 
 %    \begin{macrocode}
     \color@begingroup
     \let\@footnotetext\LT@p@ftntext
     \setlength\hsize{#1}%
     \@arrayparboxrestore
-    \vrule \@height \ht\@arstrutbox \@width \z@}
+    \everypar{%
+      \vrule \@height \ht\@arstrutbox \@width \z@
+      \everypar{}}%    
+    }
```

Comment:
Loosely related: If using paragraph hook `\AddToHook{para/begin}{...}`  instead, since `\AddToHook` acts globally, a `\RemoveFromHook` is needed in `\LT@endpbox`. Is it?

Reply:
> Loosely related: If using paragraph hook `\AddToHook{para/begin}{...}` instead, since `\AddToHook` acts globally, a `\RemoveFromHook` is needed in `\LT@endpbox`. Is it?

without having looked at the context, I think you would want to use \AddToHookNext instead (because this is what it does, but perhaps that wouldn't quite work, not sure)